### ISSUE TITLE
Calico and Canal CNIs cause complications when deploying an RKE1 cluster to Ubuntu 22.04

### ISSUE THREAD

**Issue by markusewalker** [rancher/rke#2942]
**RKE version:** 
`v1.3.11`

**Docker version: (`docker version`,`docker info` preferred)** 
`20.10.16, build aa7e414`

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)** 
Ubuntu 22.04

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
AWS

**cluster.yml file:**

**Steps to Reproduce:**
1. Provision 3 Ubuntu 22.04 AMIs to be used in a 3 node RKE cluster.
2. After properly setting up each of the AMIs to meet system requirements, run `rke config` and fill out the prompts. Ensure that you specify `Canal` as the CNI.
3. Copy the `kube_config_cluster.yml` into `$HOME/.kube/config` and verify with `kubectl get nodes` it is up and running:
```
$ kubectl get nodes
NAME           STATUS   ROLES                      AGE     VERSION
mew-ubuntu-1   Ready    controlplane,etcd,worker   7m19s   v1.23.6
mew-ubuntu-2   Ready    controlplane,etcd,worker   7m19s   v1.23.6
mew-ubuntu-3   Ready    controlplane,etcd,worker   7m19s   v1.23.6
```
4. Run `kubectl get pods -A` and observe the failures:
```
$ kubectl get pods -A
NAMESPACE       NAME                                      READY   STATUS      RESTARTS   AGE
ingress-nginx   ingress-nginx-admission-create-7dkxl      0/1     Evicted     0          2m9s
ingress-nginx   ingress-nginx-admission-create-bjdrb      0/1     Evicted     0          2m9s
ingress-nginx   ingress-nginx-admission-create-cbq55      0/1     Completed   0          2m9s
ingress-nginx   ingress-nginx-admission-create-p5vvn      0/1     Evicted     0          2m9s
ingress-nginx   ingress-nginx-admission-patch-zd8lj       0/1     Completed   0          2m9s
ingress-nginx   nginx-ingress-controller-4hrnx            0/1     Evicted     0          37s
ingress-nginx   nginx-ingress-controller-hpx7b            0/1     Evicted     0          37s
ingress-nginx   nginx-ingress-controller-rdm2h            1/1     Running     0          2m9s
kube-system     calico-kube-controllers-fc7fcb565-vtl62   1/1     Running     0          2m23s
kube-system     canal-cvt7j                               2/2     Running     0          2m24s
kube-system     canal-klkgz                               2/2     Running     0          2m24s
kube-system     canal-rr2k2                               2/2     Running     0          2m24s
kube-system     coredns-548ff45b67-7kkz2                  1/1     Running     0          2m19s
kube-system     coredns-548ff45b67-jjspc                  1/1     Running     0          2m13s
kube-system     coredns-autoscaler-d5944f655-gx5zp        0/1     Error       0          2m19s
kube-system     coredns-autoscaler-d5944f655-qx8px        1/1     Running     0          2m2s
kube-system     metrics-server-5c4895ffbd-jzx9c           1/1     Running     0          2m14s
kube-system     rke-coredns-addon-deploy-job-fqcjp        0/1     Completed   0          2m21s
kube-system     rke-ingress-controller-deploy-job-q7jjl   0/1     Completed   0          2m11s
kube-system     rke-metrics-addon-deploy-job-6pbjp        0/1     Completed   0          2m16s
kube-system     rke-network-plugin-deploy-job-6mrrx       0/1     Completed   0          2m26s
```
5. Run `rke remove --force` to destroy the cluster.
6. Repeat steps 2 through 4 TWICE, but ensure that you are using CNIs Flannel and Calico. Results below:

_**Flannel**_
```
$ kubectl get pods -A
NAMESPACE       NAME                                      READY   STATUS      RESTARTS   AGE
ingress-nginx   ingress-nginx-admission-create-7lb2q      0/1     Completed   0          49s
ingress-nginx   ingress-nginx-admission-patch-sfkh6       0/1     Completed   2          49s
ingress-nginx   nginx-ingress-controller-5gcsw            1/1     Running     0          49s
ingress-nginx   nginx-ingress-controller-7mg27            1/1     Running     0          49s
ingress-nginx   nginx-ingress-controller-qfks4            1/1     Running     0          49s
kube-system     coredns-548ff45b67-9ncqq                  1/1     Running     0          55s
kube-system     coredns-548ff45b67-zw6gq                  1/1     Running     0          60s
kube-system     coredns-autoscaler-d5944f655-pkjh6        1/1     Running     0          60s
kube-system     kube-flannel-bw9sx                        2/2     Running     0          64s
kube-system     kube-flannel-tpj7n                        2/2     Running     0          64s
kube-system     kube-flannel-vct66                        2/2     Running     0          65s
kube-system     metrics-server-5c4895ffbd-sxk5r           1/1     Running     0          54s
kube-system     rke-coredns-addon-deploy-job-2qwpz        0/1     Completed   0          61s
kube-system     rke-ingress-controller-deploy-job-sbfhl   0/1     Completed   0          51s
kube-system     rke-metrics-addon-deploy-job-qld2b        0/1     Completed   0          56s
kube-system     rke-network-plugin-deploy-job-lm9fr       0/1     Completed   0          66s
```

_**Calico**_
```
$ kubectl get pods -A
NAMESPACE       NAME                                       READY   STATUS              RESTARTS   AGE
ingress-nginx   ingress-nginx-admission-create-b9x8p       0/1     Evicted             0          72s
ingress-nginx   ingress-nginx-admission-create-jjk2g       0/1     Evicted             0          72s
ingress-nginx   ingress-nginx-admission-create-n92ln       0/1     Evicted             0          71s
ingress-nginx   ingress-nginx-admission-create-nvh2k       0/1     Evicted             0          72s
ingress-nginx   ingress-nginx-admission-create-plp9x       0/1     Evicted             0          70s
ingress-nginx   ingress-nginx-admission-create-t8nd4       0/1     Evicted             0          72s
ingress-nginx   ingress-nginx-admission-create-vtksk       0/1     Evicted             0          72s
ingress-nginx   ingress-nginx-admission-patch-64fqf        0/1     Evicted             0          72s
ingress-nginx   ingress-nginx-admission-patch-cjc4s        0/1     Evicted             0          72s
ingress-nginx   ingress-nginx-admission-patch-dxnvz        0/1     Evicted             0          72s
ingress-nginx   ingress-nginx-admission-patch-n4c46        0/1     Evicted             0          71s
ingress-nginx   ingress-nginx-admission-patch-plrk5        0/1     Evicted             0          72s
ingress-nginx   ingress-nginx-admission-patch-tjsfw        0/1     Evicted             0          71s
ingress-nginx   ingress-nginx-admission-patch-xp72v        0/1     Evicted             0          72s
ingress-nginx   nginx-ingress-controller-dwj8t             0/1     Evicted             0          9s
ingress-nginx   nginx-ingress-controller-rljxs             0/1     ContainerCreating   0          73s
ingress-nginx   nginx-ingress-controller-sdqzf             0/1     Evicted             0          9s
kube-system     calico-kube-controllers-5f7cbbd955-cwxxr   1/1     Running             0          92s
kube-system     calico-node-m9kb7                          1/1     Running             0          92s
kube-system     calico-node-mhmzq                          1/1     Running             0          92s
kube-system     calico-node-vlhqd                          1/1     Running             0          92s
kube-system     coredns-548ff45b67-8p26g                   1/1     Running             0          83s
kube-system     coredns-548ff45b67-x997k                   1/1     Running             0          79s
kube-system     coredns-autoscaler-d5944f655-687s7         1/1     Running             0          83s
kube-system     metrics-server-5c4895ffbd-2wddm            1/1     Running             0          78s
kube-system     rke-coredns-addon-deploy-job-wxl95         0/1     Completed           0          85s
kube-system     rke-ingress-controller-deploy-job-gh676    0/1     Completed           0          74s
kube-system     rke-metrics-addon-deploy-job-xbt7c         0/1     Completed           0          79s
kube-system     rke-network-plugin-deploy-job-r6wnt        0/1     Completed           0          95s
```

**Actual Behavior:**
When deploying an RKE1 cluster with CNIs canal and calico to Ubuntu 22.04, there are unexpected failures with the `ingress-nginx` and `kube-system` namespaces.

**Expected Behavior:**
There shouldn't be failures in any of the namespaces when deploying an RKE1 cluster to Ubuntu 22.04, regardless of the CNI chosen.

**Comment by markusewalker** [comment]
Ubuntu 22.04 is currently not supported and being looked into. Putting this into backlog until that is the case.

**Comment by markusewalker** [comment]
Despite this being in the backlog, when Ubuntu 22.04 is eventually supported, here is the QA test plan:

**TEST PLAN**
1. Provision 3 Ubuntu 22.04 AMIs to be used in the RKE1 cluster.
2. After properly setting up each node to meet system requirements,  run `rke config` and fill out the prompts.
     - Ensure that `Calico` is the CNI chosen.
3. Once `cluster.yml` is created, run `rke up --config cluster.yml`.
4. Copy the resulting `kube_config_cluster.yml` into `$HOME/.kube`.
5. Run `kubectl get nodes` and validate that the cluster is up and running.
6. Run `rke remove --force` and repeat steps 2-5 but using Canal as the CNI.

**Comment by kinarashah** [comment]
https://github.com/rancher/rancher/issues/38984

**Comment by markusewalker** [comment]
Validated that this is addressed in `v1.3.15-rc3`.

**ENVIRONMENT DETAILS**
- Ubuntu 22.04 AMIs
     - 3 for controplane, etcd, worker

**TEST RESULT**
**PASS**

**VERIFICATION STEPS**
1. Provisioned 3 Ubuntu 22.04 AMIs to be used in the RKE1 cluster.
2. Set up each node meeting system requirements and ran `rke config`.
     - Ensured that `Canal` is the CNI I chose.
3. Verified `cluster.yml` was created and ran `rke up --config cluster.yml`.
4. Verified `kube_config_cluster.yml` was created and moved to `$HOME/.kube`.
5. Ran `kubectl get nodes` and validated the cluster is up:
```
$ kubectl get nodes
NAME       STATUS   ROLES                      AGE     VERSION
mew1-rke   Ready    controlplane,etcd,worker   6m43s   v1.24.4
mew2-rke   Ready    controlplane,etcd,worker   6m42s   v1.24.4
mew3-rke   Ready    controlplane,etcd,worker   6m43s   v1.24.4
```
6. Run `kubectl get pods -A` and validated everything looks good:
```
$ kubectl get pods -A
NAMESPACE       NAME                                       READY   STATUS      RESTARTS   AGE
ingress-nginx   ingress-nginx-admission-create-rzxk6       0/1     Completed   0          6m34s
ingress-nginx   ingress-nginx-admission-patch-sfsdt        0/1     Completed   1          6m34s
ingress-nginx   nginx-ingress-controller-ch9tf             1/1     Running     0          6m34s
ingress-nginx   nginx-ingress-controller-sq4lr             1/1     Running     0          6m34s
ingress-nginx   nginx-ingress-controller-zntpq             1/1     Running     0          6m34s
kube-system     calico-kube-controllers-74df54cbb7-99brg   1/1     Running     0          7m3s
kube-system     canal-2b54f                                2/2     Running     0          7m3s
kube-system     canal-7qqdn                                2/2     Running     0          7m3s
kube-system     canal-cf9dw                                2/2     Running     0          7m3s
kube-system     coredns-59499769fb-dkhdv                   1/1     Running     0          6m54s
kube-system     coredns-59499769fb-vz6x4                   1/1     Running     0          6m47s
kube-system     coredns-autoscaler-67cbd4599c-dgfbb        1/1     Running     0          6m54s
kube-system     metrics-server-585b7cc746-7r4jz            1/1     Running     0          6m44s
kube-system     rke-coredns-addon-deploy-job-mqfxn         0/1     Completed   0          6m56s
kube-system     rke-ingress-controller-deploy-job-khx65    0/1     Completed   0          6m36s
kube-system     rke-metrics-addon-deploy-job-dpndr         0/1     Completed   0          6m46s
kube-system     rke-network-plugin-deploy-job-cwsbt        0/1     Completed   0          7m6s
```
7. Repeated steps 2-5, but changed the CNI to be Calico. This was successful as well:
```
$ kubectl get pods -A
NAMESPACE       NAME                                       READY   STATUS      RESTARTS   AGE
ingress-nginx   ingress-nginx-admission-create-rgvmp       0/1     Completed   0          30s
ingress-nginx   ingress-nginx-admission-patch-m9s85        0/1     Completed   0          30s
ingress-nginx   nginx-ingress-controller-lfhf7             1/1     Running     0          30s
ingress-nginx   nginx-ingress-controller-pmdl8             1/1     Running     0          30s
ingress-nginx   nginx-ingress-controller-zcs48             1/1     Running     0          30s
kube-system     calico-kube-controllers-5cc86cb94f-sg9nn   1/1     Running     0          59s
kube-system     calico-node-5mg9q                          1/1     Running     0          59s
kube-system     calico-node-8gsqz                          1/1     Running     0          59s
kube-system     calico-node-gqm9z                          1/1     Running     0          59s
kube-system     coredns-59499769fb-96mnv                   1/1     Running     0          50s
kube-system     coredns-59499769fb-kp9xc                   1/1     Running     0          49s
kube-system     coredns-autoscaler-67cbd4599c-wg25s        1/1     Running     0          50s
kube-system     metrics-server-585b7cc746-mbg7t            1/1     Running     0          40s
kube-system     rke-coredns-addon-deploy-job-c7hvh         0/1     Completed   0          52s
kube-system     rke-ingress-controller-deploy-job-cmnsd    0/1     Completed   0          32s
kube-system     rke-metrics-addon-deploy-job-lsmbk         0/1     Completed   0          42s
kube-system     rke-network-plugin-deploy-job-wmfft        0/1     Completed   0          62s
```
8. Repeated steps 2-5, but changed the CNI to be Flannel. This was successful as well:
```
$ kubectl get pods -A
NAMESPACE       NAME                                      READY   STATUS      RESTARTS   AGE
ingress-nginx   ingress-nginx-admission-create-7qkcr      0/1     Completed   0          21s
ingress-nginx   ingress-nginx-admission-patch-qcjzv       0/1     Completed   1          21s
ingress-nginx   nginx-ingress-controller-nx9d6            0/1     Running     0          21s
ingress-nginx   nginx-ingress-controller-slswx            0/1     Running     0          21s
ingress-nginx   nginx-ingress-controller-xprg2            1/1     Running     0          21s
kube-system     coredns-59499769fb-4lf6m                  1/1     Running     0          42s
kube-system     coredns-59499769fb-lwhsk                  1/1     Running     0          39s
kube-system     coredns-autoscaler-67cbd4599c-qndc8       1/1     Running     0          42s
kube-system     kube-flannel-8hq2n                        2/2     Running     0          52s
kube-system     kube-flannel-h4jzj                        2/2     Running     0          52s
kube-system     kube-flannel-p4k74                        2/2     Running     0          52s
kube-system     metrics-server-585b7cc746-ntpsd           1/1     Running     0          31s
kube-system     rke-coredns-addon-deploy-job-rnxwm        0/1     Completed   0          43s
kube-system     rke-ingress-controller-deploy-job-j7w55   0/1     Completed   0          23s
kube-system     rke-metrics-addon-deploy-job-zrbfl        0/1     Completed   0          33s
kube-system     rke-network-plugin-deploy-job-zn6j7       0/1     Completed   0          53s
```
